### ISSUE TITLE
Build fails with "Warning: Could not create settings", "java.lang.IllegalArgumentException"…

### ISSUE THREAD

**Issue by nhoizey** [GoogleChromeLabs/bubblewrap#144]
**Describe the bug**

When trying to build the TWA, it breaks with this log:

<details>
<pre>
Installing Android Build Tools. Please, read and accept the license agreement
Installing Build Tools
Warning: Could not create settings
java.lang.IllegalArgumentException
	at com.android.sdklib.tool.sdkmanager.SdkManagerCliSettings.<init>(SdkManagerCliSettings.java:428)
	at com.android.sdklib.tool.sdkmanager.SdkManagerCliSettings.createSettings(SdkManagerCliSettings.java:152)
	at com.android.sdklib.tool.sdkmanager.SdkManagerCliSettings.createSettings(SdkManagerCliSettings.java:134)
	at com.android.sdklib.tool.sdkmanager.SdkManagerCli.main(SdkManagerCli.java:57)
	at com.android.sdklib.tool.sdkmanager.SdkManagerCli.main(SdkManagerCli.java:48)
Usage:
  sdkmanager [--uninstall] [<common args>] [--package_file=<file>] [<packages>...]
  sdkmanager --update [<common args>]
  sdkmanager --list [<common args>]
  sdkmanager --licenses [<common args>]
  sdkmanager --version

With --install (optional), installs or updates packages.
    By default, the listed packages are installed or (if already installed)
    updated to the latest version.
With --uninstall, uninstall the listed packages.

    <package> is a sdk-style path (e.g. "build-tools;23.0.0" or
             "platforms;android-23").
    <package-file> is a text file where each line is a sdk-style path
                   of a package to install or uninstall.
    Multiple --package_file arguments may be specified in combination
    with explicit paths.

With --update, all installed packages are updated to the latest version.

With --list, all installed and available packages are printed out.

With --licenses, show and offer the option to accept licenses for all
     available packages that have not already been accepted.

With --version, prints the current version of sdkmanager.

Common Arguments:
    --sdk_root=<sdkRootPath>: Use the specified SDK root instead of the SDK
                              containing this tool

    --channel=<channelId>: Include packages in channels up to <channelId>.
                           Common channels are:
                           0 (Stable), 1 (Beta), 2 (Dev), and 3 (Canary).

    --include_obsolete: With --list, show obsolete packages in the
                        package listing. With --update, update obsolete
                        packages as well as non-obsolete.

    --no_https: Force all connections to use http rather than https.

    --proxy=<http | socks>: Connect via a proxy of the given type.

    --proxy_host=<IP or DNS address>: IP or DNS address of the proxy to use.

    --proxy_port=<port #>: Proxy port to connect to.

    --verbose: Enable verbose output.

* If the env var REPO_OS_OVERRIDE is set to "windows",
  "macosx", or "linux", packages will be downloaded for that OS.

cli ERROR undefined
</pre>
</details>

**To Reproduce**

Run `bubblewrap build`

**Expected behavior**

**Desktop (please complete the following information):**
 - OS: macOS
 - Version 10.15.4

**Additional context**

First two levels of my folders, if it cacn help:

```
.
├── app
│   ├── build.gradle
│   └── src
├── build.gradle
├── gradle
│   └── wrapper
├── gradle.properties
├── gradlew
├── gradlew.bat
├── jdk
│   └── Contents
├── settings.gradle
├── signing-keystore.keystore
├── store_icon.png
├── tools
│   ├── NOTICE.txt
│   ├── bin
│   ├── lib
│   └── source.properties
└── twa-manifest.json
```

Also, `signing-keystore.keystore` comes from a previous build generated by pwabuilder.com

**Comment by andreban** [comment]
Would you be able to post the entire error from the console output? Posting the content of `twa-manifest.json` may help as well.

**Comment by nhoizey** [comment]
Here is the `twa-manifest.json`:

```json
{
  "packageId": "com.esviji.twa",
  "host": "play.esviji.com",
  "name": "esviji, a free puzzle game",
  "launcherName": "esviji",
  "themeColor": "#444444",
  "navigationColor": "#000000",
  "backgroundColor": "#444444",
  "enableNotifications": false,
  "startUrl": "/?utm_source=pwa",
  "iconUrl": "https://play.esviji.com/img/esviji-logo-maskable-512.png",
  "maskableIconUrl": "https://play.esviji.com/img/esviji-logo-maskable-512.png",
  "splashScreenFadeOutDuration": 300,
  "signingKey": {
    "alias": "android",
    "path": "./signing-keystore.keystore"
  },
  "appVersionName": "1",
  "appVersionCode": 1,
  "shortcuts": [],
  "generatorApp": "bubblewrap-cli",
  "webManifestUrl": "https://play.esviji.com/manifest.webmanifest",
  "fallbackType": "customtabs",
  "appVersion": "1"
}
```

**Comment by andreban** [comment]
It seems the latest version of the Android CLI tools requiring setting --sdk-root when invoking sdkmanager. Setting ANDROID_HOME was enough for previous versions. #145  fixes this.

**Comment by nhoizey** [comment]
Should I install from master to test, or do you plan to make a release?

**Comment by andreban** [comment]
I'm planning to make a release once this PR is approved and merged.

**Comment by nhoizey** [comment]
Oh, sorry, I thought it was already merged.

I will try with the PR.

**Comment by andreban** [comment]
Just released `1.1.0`. Let me know if it works.

**Comment by nhoizey** [comment]
Sorry, I installed v1.1.0, restarted `bubblewrap init  --manifest https://play.esviji.com/manifest.webmanifest` and `bubblewrap build`, but I get the exact same error.

**Comment by andreban** [comment]
I managed to upload a broken build :/. Updated again (v1.1.1). Can you try via:

```
npx @bubblewrap/cli build
```

**Comment by nhoizey** [comment]
Looks like it worked! 👍

I have an other problem at the end of the build, I will open another issue if I can't find what's wrong.

**Comment by andreban** [comment]
Glad to hear that. What's the remaining issue? Sounds like something we can
improve if things are not seamless.

**Comment by nhoizey** [comment]
I was not using the right pass phrases… 😅
I had to find the ones PWABuilder gave me the first time I got a TWA to upload to Google Play Store.

Another issue I have now is that the `appVersionCode` is still 1, and Google Play Store refuses it because there is already one apk. Should I update `twa-manifest.json` manually before running the `build`?

**Comment by andreban** [comment]
If you run `npx @bubblewrap/cli update`, it will update appVersionCode and
appVersionName and also apply any other updates to twa-manifest.json to the
Android project.

**Comment by nhoizey** [comment]
Ok, I understand.

Is there any way to use the same version number as I put in my PWA (in the package.json)?

Can I update `twa-manifest.json` manually after `npx @bubblewrap/cli update` to make sure the right version is there?

**Comment by andreban** [comment]
Updating the version in `twa-manifest.json` won't work, `app/build.gradle` has already been generated at this point. You could manually set `appVersionName` before calling `update`. A better option is manually setting the appVersionName when invoking update though:

```
npx @bubblewrap/cli update --appVersionName="my-version-name"
```

**Comment by nhoizey** [comment]
I will set it manually when calling update then, thanks!
